client/fingerprint/bridge_linux.go

```diff
@@ -89,6 +97,14 @@ func (f *BridgeFingerprint) detect(module string) error {
 	return errs
 }
 
+func (f *BridgeFingerprint) findDir(dirname string) error {
+	if _, err := os.Stat(dirname); os.IsNotExist(err) {
+		return fmt.Errorf("failed to find %s: %v", dirname, err)
+	} else {
+		return nil
+	}
```

Comment:
This is only checking for `IsNotExist`, which results in any other kind of error being ignored. We can probably remove the `IsNotExist` call and just check for `err != nil`.

Moreover, this seems like a trivial enough check that could be done in the `detect` method directly:

```go
loadedOrBuiltinPath := fmt.Sprintf("/sys/module/%s", module)
if _, err := os.Stat(loadedOrBuiltinPath); err != nill {
    errs = multierror.Append(errs, err)
} else {
    return nil
}
```

Reply:
Makes sense. Thanks!

Reply:
On the 'no function', I'd like to keep things consistent with the way search File pushes specific errors into the multierr. That's actually how I found where the code was!